File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/candidate/CandidateJpaRepository.java
package kyonggiyo.persistence.candidate;

import kyonggiyo.domain.candidate.Candidate;
import kyonggiyo.domain.candidate.Status;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CandidateJpaRepository extends JpaRepository<Candidate, Long> {

    Slice<Candidate> findAllByStatus(Status status, Pageable pageable);

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/candidate/repository/CandidateJpaRepository.kt
package kyonggiyo.candidate.repository

import kyonggiyo.application.candidate.domain.entity.Candidate
import kyonggiyo.application.candidate.domain.vo.Status
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice
import org.springframework.data.jpa.repository.JpaRepository

interface CandidateJpaRepository : JpaRepository<Candidate, Long> {
    
    fun findAllByStatus(status: Status, pageable: Pageable): Slice<Candidate>
    
}
